apps/service_providers/llm_service/runnables.py

```diff
@@ -136,7 +146,16 @@ def invoke(self, input: str, config: RunnableConfig | None = None, *args, **kwar
             raise GenerationCancelled(result)
 
         experiment_tag = configurable.get("experiment_tag")
-        self.adapter.post_run_hook(output, save_output_to_history, experiment_tag=experiment_tag, message_metadata={})
+        self.history_manager.add_messages_to_history(
+            input=input,
+            save_input_to_history=save_input_to_history,
+            input_message_metadata={},
+            output=output,
+            save_output_to_history=save_output_to_history,
+            experiment_tag=experiment_tag,
+            output_message_metadata={},
+        )
```

Comment:
I think it's fine to save the input and output here but we may need some error handling to ensure that the input get's saved even if there's an error.